drivers/dac/dac_ad569x.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2024 Jan Kubiznak <jan.kubiznak@deveritec.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/dac.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+LOG_MODULE_REGISTER(dac_ad569x, CONFIG_DAC_LOG_LEVEL);
+
+#define DAC_AD569X_MAX_CHANNEL 0
+
+#define AD569X_CTRL_GAIN(x)  (FIELD_PREP(BIT(11), x))
+#define AD569X_CTRL_REF(x)   (FIELD_PREP(BIT(12), x))
+#define AD569X_CTRL_PD(x)    (FIELD_PREP(BIT_MASK(2) << 13, x))
+#define AD569X_CTRL_RESET(x) (FIELD_PREP(BIT(15), x))
+
+enum ad569x_command {
+	AD569X_CMD_WRITE = 0x10,
+	AD569X_CMD_UPDATE = 0x20,
+	AD569X_CMD_WRITE_AND_UPDATE = 0x30,
+	AD569X_CMD_CONFIGURE = 0x40,
+};
+
+struct ad569x_config {
+	struct i2c_dt_spec bus;
+	uint8_t resolution;
+	uint8_t gain;
+	uint8_t voltage_reference;
+	uint8_t power_down_mode;
+};
+
+enum {
+	AD569X_CTRL_NO_RESET = 0,
+	AD569X_CTRL_PERFORM_RESET = 1,
+	AD569X_CTRL_RESET_DEFAULT = AD569X_CTRL_NO_RESET
+};
+
+static int ad569x_write(const struct device *dev, enum ad569x_command command, uint16_t value)
+{
+	const struct ad569x_config *config = dev->config;
+
+	uint8_t tx_data[3];
+
+	tx_data[0] = command;
+	sys_put_be16(value, tx_data + 1);
+
+	return i2c_write_dt(&config->bus, tx_data, sizeof(tx_data));
+}
+
+static int ad569x_read(const struct device *dev, uint16_t *value)
+{
+	const struct ad569x_config *config = dev->config;
+
+	uint8_t rx_data[2];
+	int result;
```

Comment:
nit: all dac drivers but one use `ret` for the return value, I suggest changing it in this file, it's shorter and still unambiguous